source/components/CurrencyInput/CurrencyInput.js

```diff
@@ -1,75 +1,91 @@
 import classnames from 'classnames'
 import { omit } from 'ramda'
 import React, { Component } from 'react'
+import NumberFormat from 'react-number-format'
 import { debounce } from '../../utils'
 import './CurrencyInput.css'
 
-let isCurrencyPrefixed = language =>
-	!!Intl.NumberFormat(language, {
+let currencyFormat = language => ({
+	isCurrencyPrefixed: !!Intl.NumberFormat(language, {
 		style: 'currency',
 		currency: 'EUR'
 	})
 		.format(12)
-		.match(/€.*12/)
+		.match(/€.*12/),
+
+	thousandSeparator: Intl.NumberFormat(language)
+		.format(1000)
+		.charAt(1),
+
+	decimalSeparator: Intl.NumberFormat(language)
+		.format(0.1)
+		.charAt(1)
+})
 
 class CurrencyInput extends Component {
 	state = {
-		value: this.props.storeValue
+		value: this.props.value || '',
+		valueHasChanged: false,
 	}
+
 	onChange = this.props.debounce
 		? debounce(this.props.debounce, this.props.onChange)
 		: this.props.onChange
-	input = React.createRef()
 
+	handleNextChange = false
+	value = undefined
 	handleChange = event => {
-		let value = event.target.value
-		value = value
-			.replace(/,/g, '.')
-			.replace(/[^\d.]/g, '')
-			.replace(/\.(.*)\.(.*)/g, '$1.$2')
-		this.setState({ value })
-
-		if (value.endsWith('.')) {
+		// Only trigger the `onChange` event if the value has changed -- and not
+		// only its formating, we don't want to call it when a dot is added in `12.`
+		// for instance
+		if (!this.handleNextChange) {
 			return
 		}
-		event.target.value = value
-
-		if (event.persist) {
-			event.persist()
+		this.handleNextChange = false
+		event.persist()
+		event.target = {
+			...event.target,
+			value: this.value
 		}
 		this.onChange(event)
 	}
-	componentDidUpdate(prevProps) {
-		if (
-			prevProps.storeValue !== this.props.storeValue &&
-			this.props.storeValue !== this.state.value
-		) {
-			this.setState({ value: this.props.storeValue })
-		}
-	}
 
```

Comment:
```suggestion
	static getDerivedStateFromProps(props, state) {
		if (props.value !== state.prevPropsValue && props.value !== state.value) {
			return {
				value: props.value,
				prevPropsValue: props.value
			}
		}
		return {
			prevPropsValue: props.value
		}
	}
```

(cf commentaire https://github.com/betagouv/syso/pull/551/files#r287820800)